### ISSUE TITLE
Removing parentheses for decorators causes TS1497 error.

### ISSUE THREAD

**Issue by erayhanoglu** [prettier/prettier#16429]
TypeScript announces [Stricter Parsing for Decorators](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#stricter-parsing-for-decorators) in v5.5. Prettier removes parentheses for decorators after commit #11849. This breaks the code and tsc throws "TS1497: Expression must be enclosed in parentheses to be used as a decorator".

**Prettier 3.3.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAFPAzjdBKAOgA90ByAcwBsIAjAQ0tMIE988AdKVbXFsq2gyYFWHKGEp0sWAAQANGcAC+IADQgIABxgBLaFmSg6AJ2MQA7gAUTCAygbm6zA+prG6YANZwYAZToAtnAAMjpQcMgAZgxYcK7uXj6+mh5h5MgwxgCucSCxAToZ2blwRJpwxjpBsAwAKhVQJjpwdtGUsepYaZRwAIpZEPBRMbkAVlhEvt19A0NIbR0gAI6zcJZmmnYgUgC04XAAJodqIJl0OpRpAMIQAQF0yNuUlCddUFRwAIIwmTo0WfBLBVQuFhu1cgALGABSgAdQhOmwKTAcF8tkROgAbojmI8wNITpicgBJKBHWC+MCVbSfMm+GDMHpgxaaMyxWHuTSPVktCqYiLqMKxYwwdZ0cj3Zm5FLGYWPBnlLBUnTaE6ssIwWE6A4wCHIAAcAAZ1MY4CsdKaxRKHvMRuoYHQaFqdXqkAAmdRZWK1R2tO0gOABGiHI4HYJ0d5ZcVwABiEGM9x+aUedABEBASiUQA)

```sh
--parser typescript
```

**Input:**

```tsx
@(test().x('global').y())
@test().y('global').y()
class X {}
```

**Output:**

```tsx
@test().x("global").y()
@test().y("global").y()
class X {}

```

**Expected behavior:**

**Input:**

```tsx
@(test().x('global').y())
@test().y('global').y()
class X {}
```

**Output:**

```tsx
@(test().x("global").y())
@(test().y("global").y())
class X {}

```

**Comment by kachkaev** [comment]
Possibly relevant to #15611

**Comment by fisker** [comment]
Let's add them back.

**Comment by ujwal-setlur** [comment]
Any updates on this?

**Comment by ujwal-setlur** [comment]
This is pretty critical IMHO. This issue prevents anyone from using typescript 5.5 with decorator. A speedy resolution is much appreciated.

**Comment by fisker** [comment]
Fixed by  #16458
